### PR TITLE
pkg/mount: optimizations

### PR DESCRIPTION
**Pull request by kolyshkin**
Various minor things here and there to streamline pkg/mount operations.

### CODE REVIEW

File: pkg/mount/mount.go [hunk 1]
```diff
@@ -102,21 +102,12 @@ func Mounted(mountpoint string) (bool, error) {
...
 		}
 	}
-	return ForceMount(device, target, mType, options)
-}
-
-// ForceMount will mount a filesystem according to the specified configuration,
-// *regardless* if the target path is not already mounted. Options must be
-// specified like the mount or fstab unix commands: "opt1=val1,opt2=val2". See
-// flags.go for supported option flags.
-func ForceMount(device, target, mType, options string) error {
```
Comment (thaJeztah) [inline pkg/mount/mount.go]: I see there's a couple of uses of this function in the wild; https://github.com/search?l=Go&q=%22mount.ForceMount%28%22&type=Code

Should we keep the wrapper and add a `deprecated` to the doc?

```
// ForceMount will mount a filesystem according to the specified configuration,
// *regardless* if the target path is not already mounted. Options must be
// specified like the mount or fstab unix commands: "opt1=val1,opt2=val2". See
// flags.go for supported option flags.
// Deprecated: use Mount() instead
```

File: pkg/mount/sharedsubtree_linux.go [hunk 2]
```diff
@@ -67,5 +67,5 @@ func ensureMountedAs(mountPoint, options string) error {
 		return err
 	}
 
-	return ForceMount("", mountPoint, "none", options)
+	return mount("", mountPoint, uintptr(0), options)
```
Comment (thaJeztah) [inline pkg/mount/sharedsubtree_linux.go]: This is failing;

```
00:41:38 pkg/mount/sharedsubtree_linux.go:70:31:warning: cannot use uintptr(0) (constant 0 of type uintptr) as string value in argument to mount (gosimple)
00:41:38 pkg/mount/sharedsubtree_linux.go:70:50:warning: too few arguments in call to mount (unconvert)
00:41:38 pkg/mount/sharedsubtree_linux.go:70:43:warning: cannot use options (variable of type string) as uintptr value in argument to mount (unconvert)
00:41:38 pkg/mount/sharedsubtree_linux.go:70:31:warning: cannot use uintptr(0) (constant 0 of type uintptr) as string value in argument to mount (unconvert)
```

**Review by thaJeztah — CHANGES_REQUESTED** [review]

**Comment by kolyshkin** [comment]
OK, let the `ForceMount` stay as it is... A new version pushed doing same optimizations without removing `ForceMount`.

File: pkg/mount/sharedsubtree_linux.go (hunk 2 above)
Comment (kolyshkin) [reply pkg/mount/sharedsubtree_linux.go]: Thanks, I was too soon to push this. Please see the new commits.

File: pkg/mount/mount.go (hunk 1 above)
Comment (kolyshkin) [reply pkg/mount/mount.go]: I should have checked if there are any external users... My new commits leave this function as it was.

**Comment by kolyshkin** [comment]
OK it's too late now, need to take a fresh look tomorrow; setting WIP for now

**Comment by kolyshkin** [comment]
Some night thoughs:

1. `MakeMount` (I blame myself for choosing such a bad name! should have been `MakeBindMount`) does essentially the same thing as Mount()
2. ensureMountedAs does the same thing as Mount(), too

Need to carefully look at it and minimize the code flow.

**Comment by kolyshkin** [comment]
OK I think this one is good to go. @cpuguy83 PTAL

**Review by thaJeztah — APPROVED** [review]
SGTM

ping @cpuguy83

**Comment by thaJeztah** [comment]
ping @cpuguy83 PTAL

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Comment by olljanat** [comment]
Derek add label: rebuild/*

**Pull request merged**